src/notification.rs

```diff
@@ -10,68 +15,132 @@ pub use self::toast::prepare;
+            notified_times: HashMap::new(),
+        }
+    }
 
-pub fn direct_message(config: &config::Notifications<Sound>, nick: NickRef) {
-    show_notification(
-        &config.direct_message,
-        "Direct message",
-        format!("{} sent you a direct message", nick),
-    );
-}
+    pub fn notify(
+        &mut self,
+        config: &config::Notifications<Sound>,
+        notification: &Notification,
+        server: Option<impl ToString>,
+    ) {
+        if let Some(server) = server {
+            match notification {
+                Notification::Connected => {
+                    self._execute(
+                        &config.connected,
+                        notification,
+                        "Connected",
+                        server.to_string(),
+                    );
+                }
+                Notification::Disconnected => {
+                    self._execute(
+                        &config.disconnected,
+                        notification,
+                        "Disconnected",
+                        server.to_string(),
+                    );
+                }
+                Notification::Reconnected => {
+                    self._execute(
+                        &config.reconnected,
+                        notification,
+                        "Reconnected",
+                        server.to_string(),
+                    );
+                }
+                Notification::MonitoredOnline(targets) => {
+                    targets.iter().for_each(|target| {
+                        self._execute(
+                            &config.monitored_online,
+                            notification,
+                            &format!("{} is online", target.nickname()),
+                            server.to_string(),
+                        );
+                    });
+                }
+                Notification::MonitoredOffline(targets) => {
+                    targets.iter().for_each(|target| {
+                        self._execute(
+                            &config.monitored_offline,
+                            notification,
+                            &format!("{} is offline", target),
+                            server.to_string(),
+                        );
+                    });
+                }
+                Notification::FileTransferRequest(nick) => {
+                    self._execute(
+                        &config.file_transfer_request,
+                        notification,
+                        &format!("File transfer from {}", nick),
+                        server.to_string(),
+                    );
+                }
+                _ => {}
+            }
+        }
 
-pub fn highlight(config: &config::Notifications<Sound>, nick: NickRef, channel: String) {
-    show_notification(
-        &config.highlight,
-        "Highlight",
-        format!("{} highlighted you in {}", nick, channel),
-    );
-}
+        match notification {
```

Comment:
I don't like we have two match statements, and they end with `_ => {}`. When we then add new notifications it could be hard to remember to add them here, instead of relying on the compiler telling us we are missing an arm.

Instead, perhaps do something like in a simple `match notification {}`

```rust
match notification {
        Notification::Connected => {
            if let Some(server) = server{
                self._execute(&config.connected, notification, "Connected", server.clone());
            }
        }
[....]
}
```

Reply:
That's a very good point. I modified this to your suggestion :pray: